protoc-gen-twirp/generator.go

```diff
@@ -644,6 +644,14 @@ func (t *twirp) generateUtils() {
 	t.P(`func (e *internalWithCause) WithMeta(key string, val string) `, t.pkgs["twirp"], `.Error { return e }`)
 	t.P()
 
+	t.P(`// badRequestError is used when the twirp server cannot unmarshal a request`)
+	t.P(`func badRequestError(cause error, msg string) `, t.pkgs["twirp"], `.Error {`)
+	t.P(`	err := `, t.pkgs["twirp"], `.NewError(`, t.pkgs["twirp"], `.InvalidArgument, msg)`)
+	t.P(`	err = err.WithMeta("twirp_invalid_request", cause.Error())`)
```

Comment:
I think being able to see the error is important to know what character was misspelled, so it is easy to correct. But we should double check if exposing the Go error can introduce security risks ?

Reply:
I agree, this might be leaking more information about the underlying system (i.e. json/protobuf decoding version) than is really necessary. A generic message along the lines of `the [protobuf/json] message could not be decoded` points the user in the right direction without posing a security risk.